### PR TITLE
MULE-14035: Automatic retry on remote Closed exception does not respe…

### PR DESCRIPTION
**Pull request by facundovs**
…ct RFC 7320.

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 1]
```diff
@@ -87,8 +86,10 @@
     static final String WORKSTATION_DEBUG = "Workstation";
     static final String AUTHENTICATION_TYPE_DEBUG = "Authentication Type";
     static final String QUERY_PARAMS_DEBUG = "Query Params";
-    private static final String REMOTELY_CLOSED = "Remotely closed";
-    private static final int DEFAULT_RETRY_ATTEMPTS = 3;
+    public static String RETRY_ATTEMPTS_PROPERTY = SYSTEM_PROPERTY_PREFIX + "http.client.requestAttempts";
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Need to be public?

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 2]
```diff
@@ -87,8 +86,10 @@
...
+    public static final String REMOTELY_CLOSED = "Remotely closed";
+    public static final int DEFAULT_RETRY_ATTEMPTS = 3;
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Idem

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 3]
```diff
@@ -87,8 +86,10 @@
...
+    private static int RETRY_ATTEMPTS ;
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Why is this defined as a static constant when you are setting/resetting it for each new instance? Should this be a non-static field?

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 4]
```diff
@@ -144,6 +145,7 @@ public void initialise() throws InitialisationException
                                       requestStreamingMode, sendBodyMode, parseResponse, responseTimeout);
 
         notificationHelper = new NotificationHelper(muleContext.getNotificationManager(), ConnectorMessageNotification.class, false );
+        setRetryAttempts();
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Is extracting this into a method needed, when it's simply assigning retryCount

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 5]
```diff
@@ -772,4 +774,9 @@ private FieldDebugInfo getSecurityFieldDebugInfo(MuleEvent event)
 
         return securityFieldDebugInfo;
     }
+
+    private void setRetryAttempts ()
+    {
+        RETRY_ATTEMPTS = getInteger(RETRY_ATTEMPTS_PROPERTY, DEFAULT_RETRY_ATTEMPTS);
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Read system property statically to constant.

File: modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java [hunk 6]
```diff
@@ -87,8 +86,10 @@
     static final String WORKSTATION_DEBUG = "Workstation";
     static final String AUTHENTICATION_TYPE_DEBUG = "Authentication Type";
     static final String QUERY_PARAMS_DEBUG = "Query Params";
-    private static final String REMOTELY_CLOSED = "Remotely closed";
-    private static final int DEFAULT_RETRY_ATTEMPTS = 3;
+    static String RETRY_ATTEMPTS_PROPERTY = SYSTEM_PROPERTY_PREFIX + "http.client.requestAttempts";
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: -> maxRetry?

Comment (dfeist) [reply modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: client or requester? Are there any other system properties already to consider consistency?

Comment (facundovs) [reply modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequester.java]: Well, I could see the http.client.headerSectionSize system property.

**Pull request merged**